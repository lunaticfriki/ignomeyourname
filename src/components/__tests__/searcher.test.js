import React from 'react'
import { Provider } from 'react-redux'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import Searcher from '../searcher'
import store from '../../store/index'

describe('Searcher component suite', () => {
  const wrapperMount = mount(
    <Provider store={store}>
      <Searcher />
    </Provider>
  )
  const dummyProps = {
    gnomes: [],
    setGnome: jest.fn(),
  }
  test('Searcher snapshot', () => {
    const component = renderer.create(
      <Provider store={store}>
        <Searcher />
      </Provider>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('Should trigger a function on submit to update', () => {
    const button = wrapperMount.find('button')
    button.simulate('click')
    expect(dummyProps.setGnome.mock.calls).toEqual([])
  })
})
