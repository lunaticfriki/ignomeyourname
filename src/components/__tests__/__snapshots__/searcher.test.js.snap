// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component suite Searcher snapshot 1`] = `
<div
  className="sc-AxjAm dpccUt"
>
  <form
    onSubmit={[Function]}
  >
    <label
      className="c-use-gnome"
      htmlFor="use-gnome-findyour gnome"
    >
      Find your gnome
      <select
        disabled={true}
        id="use-gnome-findyour gnome"
        onBlur={[Function]}
        onChange={[Function]}
        value=""
      >
        <option
          hidden={true}
        >
           Choose your gnome!
        </option>
      </select>
    </label>
    <button>
      Search!
    </button>
  </form>
</div>
`;
